design/nfs/nfs-controller-runtime.md

```diff
@@ -0,0 +1,161 @@
+# Implement controller-runtime in Rook NFS Operator
+
+## Background
+
+This proposal is to implement controller-runtime in Rook NFS Operator to improve reliability of the operator itself. Currently, Rook nfs-operator only simply watches an event of CustomResource from an informer using simple WatchCR method which has limited functionality such as event can not be re-queued if failed. To implement controller-runtime is expected to overcome the shortcomings of current implementation.
```

Comment:
To provide more useful context, we could link to the WatchCR function definition.  That way people will be able to easily click and see what is being referred to.

https://github.com/rook/rook/blob/release-1.3/pkg/operator/k8sutil/customresource.go#L48